lua/nvim-treesitter/parsers.lua

```diff
@@ -1,7 +1,31 @@
 local api = vim.api
 local ts = vim.treesitter
 
-local filetype_to_parsername = {
+local new_lang_api = ts.language.register ~= nil
+
+local filetype_to_parsername = {}
+
+if new_lang_api then
+  filetype_to_parsername = setmetatable({}, {
+    __newindex = function(_, k, v)
+      require("nvim-treesitter.utils").notify(
+        "filetype_to_parsername is deprecated, please use 'vim.treesitter.language.register'",
+        vim.log.levels.WARN
+      )
```

Comment:
this warning ☝️

Reply:
This warning only appears if `vim.treesitter.language.register` is present, meaning you are running vim v0.9.

Reply:
No, I mean it's been around long enough, and using `filetype_to_parsername` can be turned into a hard error. (No need to carry around deprecations forever.)

Reply:
I thought `filetype_to_parsername` was the recommended way for nvim v0.8 which replaces `used_by` in the parser-config, which warning was removed in this PR.

Reply:
This has been removed:
```lua
        __newindex = function(parserconfigtable, key, value)
          if key == "used_by" then
            require("nvim-treesitter.utils").notify(
              "used_by is deprecated, please use 'filetype_to_parsername'",
              vim.log.levels.WARN
            )
            filetype_to_parsername[value] = parsername
          else
            rawset(parserconfigtable, key, value)
          end
        end,

```

Reply:
Yes, that was a clear reading fail on my end; I didn't notice the changed message, just saw the deprecation as being moved. 🤦